ThingIFSDK/ThingIFSDK/Predicate.swift

```diff
@@ -9,7 +9,7 @@
 import Foundation
 
 /** Protocol represents Predicate */
-public protocol Predicate :  NSCoding {
+public protocol Predicate: class, NSCoding {
```

Comment:
@yongpingchen We will not use weak for predicate in Trigger. I assume we will not use weak for predicate in our SDK but developers may use weak for predicate in their applications.